File: Nivel_2/Retos/ValorMaxEnLista/ValorMaxEnLista.py
def MaxValue(lista):
    if not lista:  # más pythonic que lista == []
        return None
    else:
        max_val = lista[0]
        for valor in lista:
            if valor > max_val:
                max_val = valor
        return max_val

lista = [2, 1, 4, 3, 5]
max_value = MaxValue(lista)
if max_value is None:
    print("La lista está vacía.")
else:
    print(f"El valor máximo en la lista es: {max_value}")

File: Nivel_2/Retos/ValorMaxEnLista/ValorMaxEnLista.java
public class ValorMaxEnLista{
    public static void main(String[] args) {
        System.out.println(MaxValue(new int[]{1, 3, 4, 2, 5})); // 5
    }

    public static int MaxValue(int[] list){
        if(list == null || list.length < 1){
            throw new IllegalArgumentException("La lista no puede ser nula o vacía");
        }
        int max = list[0];
        for(int i = 1; i < list.length; i++){
            if(list[i] > max){
                max = list[i];
            }
        }
        return max;
    }
}
